src/sentry/api/endpoints/user_notification_details.py

```diff
@@ -1,14 +1,46 @@
 from __future__ import absolute_import
 
+import six
+
 from collections import defaultdict
 
 from sentry.api.bases.user import UserEndpoint
-from sentry.models import UserOption
+from sentry.models import UserOption, UserOptionValue
 
 from sentry.api.serializers import serialize, Serializer
 
 from rest_framework.response import Response
 
+from rest_framework import serializers
+
+USER_OPTION_SETTINGS = {
+    'deployNotifications': {
+        'key': 'deploy-emails',
+        'default': UserOptionValue.committed_deploys_only,  # '3'
+        'type': int,
+    },
+    'personalActivityNotifications': {
+        'key': 'self_notifications',
+        'default': UserOptionValue.all_conversations,  # '0'
+        'type': bool,
+    },
+    'selfAssignOnResolve': {
+        'key': 'self_assign_issue',
+        'default': UserOptionValue.all_conversations,  # '0'
+        'type': bool,
+    },
+    'subscribeByDefault': {
+        'key': 'subscribe_by_default',
+        'default': UserOptionValue.participating_only,  # '1'
+        'type': bool,
+    },
+    'workflowNotifications': {
+        'key': 'workflow:notifications',
+        'default': UserOptionValue.all_conversations,  # '0'
+        'type': int,
+    }
+}
```

Comment:
it's slightly sad we have these constants defined twice :x we don't have a great solution today for sharing constants between python and JS though, but we could implement one in the build step.